OpenRA.Mods.TS/Traits/SupportPowers/AttackOrderPower.cs

```diff
@@ -0,0 +1,148 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.TS.Traits
+{
+	class AttackOrderPowerInfo : SupportPowerInfo, Requires<AttackBaseInfo>
+	{
+		public override object Create(ActorInitializer init) { return new AttackOrderPower(init.Self, this); }
+	}
+
+	class AttackOrderPower : SupportPower, INotifyAddedToWorld, INotifyBurstComplete
+	{
+		readonly AttackOrderPowerInfo info;
+		AttackBase attack;
+
+		public AttackOrderPower(Actor self, AttackOrderPowerInfo info)
+			: base(self, info)
+		{
+			this.info = info;
+		}
+
+		public override void SelectTarget(Actor self, string order, SupportPowerManager manager)
+		{
+			Game.Sound.PlayToPlayer(manager.Self.Owner, Info.SelectTargetSound);
+			self.World.OrderGenerator = new SelectAttackPowerTarget(self, order, manager, info.Cursor, MouseButton.Left, attack);
+		}
+
+		public override void Activate(Actor self, Order order, SupportPowerManager manager)
+		{
+			base.Activate(self, order, manager);
+			attack.AttackTarget(Target.FromCell(self.World, order.TargetLocation), false, false, true);
+		}
+
+		void INotifyAddedToWorld.AddedToWorld(Actor self)
```

Comment:
INotifyCreated, please.  This trait isn't going to change between creation and multiple add/remove from worlds.